tls/s2n_kex.c

```diff
@@ -78,38 +80,58 @@ static int s2n_check_ecdhe(const struct s2n_cipher_suite *cipher_suite, struct s
 
 static int s2n_check_kem(const struct s2n_cipher_suite *cipher_suite, struct s2n_connection *conn)
 {
+    /* There is no support for PQ KEMs while in FIPS mode */
+    if (s2n_is_in_fips_mode()) {
+        return 0;
+    }
+
+    const struct s2n_cipher_preferences *server_cipher_preferences = NULL;
+    /* If the cipher preferences have no supported KEMs, return false. */
+    if (s2n_connection_get_cipher_preferences(conn, &server_cipher_preferences) != 0) { return 0; }
```

Comment:
Think this line should be a `GUARD()`. If the connection has `NULL` cipher_preferences, something is definitely wrong and we should bail.

Reply:
The convention for the `s2n_check_kem()` function is to return 0 if the kem is invalid, so check as written will cause us to bail.